src/storage/kv/indexer.rs

```diff
@@ -50,3 +50,82 @@ impl Indexer {
         self.index.version()
     }
 }
+
+#[derive(Clone)]
+pub struct DiskIndexEntry {
+    segment_id: u64,
+    value_offset: u64,
+    value_len: usize,
+    metadata: Option<Metadata>,
+    inlined_value: Option<Bytes>,
+}
+
+#[derive(Clone)]
+pub struct MemIndexEntry {
+    metadata: Option<Metadata>,
+    value: Bytes,
+}
+
+#[derive(Clone)]
+pub enum IndexValue {
```

Comment:
Given that you are getting rid of the `Value` trait, and all the functions for IndexValue are private (and no one can call any function over it)

1) We make this private to the crate, and consequently make the entire Snapshot and its functions private to the crate

2) We keep the Value trait (removing unnecessary functions like ts etc) and implement it on IndexValue 

I am leaning towards 1, we can make Snapshot public later if needed. LMKWYT

Reply:
I'm also fine with the Option 1.

Reply:
`Snapshot` and `IndexValue` are now private to the crate.